server/main.lua

```diff
@@ -409,23 +427,35 @@ RegisterNetEvent('qbx_vehicleshop:server:sellfinanceVehicle', function(downPayme
         }
     })
 
-    TriggerClientEvent('qbx_vehicleshop:client:buyShowroomVehicle', target.PlayerData.source, vehicle, plate, vehicleId)
+    local shopId = lib.callback.await('qbx_vehicleshop:client:getPlayerCurrentShop', target.PlayerData.source)
+    local shop = sharedConfig.shops[shopId]
+    if not shop then return end
+
+    spawnVehicle(src, {
+        coords = shop.vehicleSpawn,
+        vehicleId = vehicleId
+    })
 end)
 
 -- Check if payment is due
 RegisterNetEvent('qbx_vehicleshop:server:checkFinance', function()
     local src = source
     local player = exports.qbx_core:GetPlayer(src)
-    local result = FetchFinancedVehicleEntitiesByCitizenId(player.PlayerData.citizenid)
+    local finance = require 'server.finance'
+    local result = finance.fetchFinancedVehicleEntitiesByCitizenId(player.PlayerData.citizenid)
     if not result[1] then return end
 
     exports.qbx_core:Notify(src, locale('general.paymentduein', config.finance.paymentWarning))
     Wait(config.finance.paymentWarning * 60000)
-    local vehicles = FetchFinancedVehicleEntitiesByCitizenId(player.PlayerData.citizenid)
-    for _, v in pairs(vehicles) do
+    local vehicles = finance.fetchFinancedVehicleEntitiesByCitizenId(player.PlayerData.citizenid)
+    for _, v in ipairs(vehicles) do
         local plate = v.plate
-        exports.qbx_vehicles:DeletePlayerVehicles('vehicleId', v.id)
-        --MySQL.update('UPDATE player_vehicles SET citizenid = ? WHERE id = ?', {'REPO-'..v.citizenid, v.id}) -- Use this if you don't want them to be deleted
+        if config.deleteUnpaidFinancedVehicle then
+            exports.qbx_vehicles:DeletePlayerVehicles('vehicleId', v.id)
+        else
+            MySQL.update('UPDATE player_vehicles SET citizenid = ? WHERE id = ?', {'REPO-'..v.citizenid, v.id}) -- Use this if you don't want them to be deleted
```

Comment:
This will not work as player_vehicles' citizenid must be either nil or real.

The next best alternative is to use https://docs.qbox.re/resources/qbx_vehicles/exports/server#setplayervehicleowner to set the citizenid to nil effectively making the vehicle unowned, but can of course still be searched up by plate in the database.

Reply:
> This will not work as player_vehicles' citizenid must be either nil or real.

You mean in db datatype?

I think purpose of this is ability to restore if it get repoed, if we set citizenid to nil, then we wouldn't know who is the owner so we can restore it

Reply:
For that feature, we'd need a new column to store the former citizenid owner. Not sure if this feature is super important though.

Reply:
the table constraints have a foreign key relationship tot he player table, so invalid citizenids will not work. Like I said, the best thing to do is simply to change the owner of the vehicle to nil. In practice, if a server operator needs to revert, the player reporting will tell them the vehicle plate that was repo'd.